kamon-scala-future/src/main/scala/kamon/instrumentation/futures/scala/ScalaFutureInstrumentation.scala

```diff
@@ -28,26 +32,99 @@ object ScalaFutureInstrumentation {
   val Component = "scala.future"
 
   /**
-    * Traces the execution of a block of code with a Delayed Span which takes into account the scheduling time when run
+    * Traces the execution of an asynchronous computation modeled by a Future. This function will create a Span when it
+    * starts executing and set that Span as current while the code that created the actual Future executes and then,
+    * automatically finish that Span when the created Future finishes execution.
+    *
+    * The purpose of this function is to make it easy to trace an asynchronous computation as a whole, even though such
+    * computation might be the result of applying several transformations to an initial Future. If you are interested in
+    * tracing the intermediate computations as well, take a look at the `traceBody` and `traceFun` functions bellow.
+    */
+  def trace[T](operationName: String, tags: TagSet = TagSet.Empty, metricTags: TagSet = TagSet.Empty)(future: => Future[T])
+      (implicit settings: Settings): Future[T] = {
+
+    val spanBuilder = Kamon.internalSpanBuilder(operationName, Component)
+    if(!settings.trackMetrics) spanBuilder.doNotTrackMetrics()
+    if(tags.nonEmpty()) spanBuilder.tag(tags)
+    if(metricTags.nonEmpty()) spanBuilder.tagMetrics(metricTags)
+
+    trace(spanBuilder)(future)(settings)
+  }
+
+  /**
+    * Traces the execution of an asynchronous computation modeled by a Future. This function will create a Span when it
+    * starts executing and set that Span as current while the code that created the actual Future executes and then,
+    * automatically finish that Span when the created Future finishes execution.
+    *
+    * The purpose of this function is to make it easy to trace an asynchronous computation as a whole, even though such
+    * computation might be the result of applying several transformations to an initial Future. If you are interested in
+    * tracing the intermediate computations as well, take a look at the `traceBody` and `traceFun` functions bellow.
+    */
+  def trace[T](spanBuilder: SpanBuilder)(future: => Future[T])(implicit settings: Settings): Future[T] = {
+    if(settings.trackMetrics)
+      spanBuilder.trackMetrics()
+    else
+      spanBuilder.doNotTrackMetrics()
+
+    val futureSpan = spanBuilder.start()
+    val evaluatedFuture = Kamon.runWithSpan(futureSpan, finishSpan = false)(future)
```

Comment:
What are the chances of getting this to work with delayed spans? It doesn't look so simple at first glance, but I you would know better.

Reply:
> I imagine the main issue here would be setting the delay and starting the span when the future is actually scheduled by the ExecutorService. Not sure how easy or hard it would be to do with instrumentation.

I'm not really sure if this would be possible via instrumentation alone, that's why we ended up with an implementation that requires a bit of manual intervention + instrumentation! Also, as you mentioned above, there is no way to know if the Future has already been scheduled or not, although it wouldn't make sense to trace it if it already started since it might have already finished by the time we start tracing it.

Regarding whether it would be possible to use delayed spans here, probably we can find a way! That being said, the most common pattern I see when people use Futures is to have some code that creates the Future and then map/flatmap it a few times before completing and any delayed span using the `.trace` function would only account for the very first delay, all intermediate ones would require `.traceFunc`. This is something we should definitely discuss further, although outside of the scope of this PR.

Reply:
Sounds good! Also, I hadn't thought of the transformations. Indeed, that's a bit tricky. Happy to discuss further after 2.0 is released.